### ISSUE TITLE
Trying to reserve more bytes than the bounded queue capacity

### ISSUE THREAD

**Issue by jkawnkwh85** [odygrd/quill#215]
I got the above error message. Does that mean my log message is too long so that the ring buffer can not hold the message?

**Comment by odygrd** [comment]
yes, the way it works is that there is a bounded queue (ring buffer) with the capacity as template parameter.

When the existing ring buffer is full another new empty ring buffer gets allocated to emplace there the message. However, the capacity of the queue (ring buffer) is always the same as the one specified during compilation and the log message cannot exceed the capacity of the bounded buffer.

When you get that assertion https://github.com/odygrd/quill/blob/master/quill/include/quill/detail/spsc_queue/UnboundedQueue.h#L118 it means that a new empty ring buffer was allocated but the message still didn't fit.

The default capacity of the buffer is `131'072`. 
You can change the size if you haven't changed it already to a smaller size here and it requires to recompile the logger (https://github.com/odygrd/quill/blob/master/quill/include/quill/TweakMe.h#L141)

**Comment by odygrd** [comment]
This seemed like a limitation to me so there is a fix for it in master and also `v2.6.0`.
The unbounded queue will now resize to a capacity that is at least the size of the message.

The initial size of the unbounded queue is also configurable in runtime now and no longer needed to recompile.

```c++
  quill::Config cfg;
  cfg.default_queue_capacity = 65'536
  quill::configure(cfg);
  quill::start();
````

**Comment by jkawnkwh85** [comment]
Cool! thanks for the detailed explanation and the improvement!
